test/SuperResponsiveTable.test.js

```diff
@@ -79,6 +94,110 @@ describe('SuperResponsiveTable', () => {
 
     expect(getTdBefore.length).toBe(3)
   })
+
+  it('does not call injectBaseStyles on mount', () => {
+    jest.spyOn(baseStyleUtils, 'injectBaseStyles')
+    setup()
+
+    expect(baseStyleUtils.injectBaseStyles).not.toHaveBeenCalled()
+  })
+
+  describe('withBaseStyles is truthy', () => {
+    it('calls injectBaseStyles on mount', () => {
+      jest.spyOn(baseStyleUtils, 'injectBaseStyles')
+      setup(
+        <Table withBaseStyles>
+          <Thead>
+            <Tr>
+              <Th>Header 1</Th>
+              <Th>Header 2</Th>
+              <Th>Header 3</Th>
+            </Tr>
+          </Thead>
+          <Tbody>
+            <Tr>
+              <Td>Row 1</Td>
+              <Td>Row 2</Td>
+              <Td>Row 3</Td>
+            </Tr>
+          </Tbody>
+        </Table>
+      )
+
+      expect(baseStyleUtils.injectBaseStyles).toHaveBeenCalledTimes(1)
+    })
+
+    it('calls injectBaseStyles again if props have changed', () => {
+      jest.spyOn(baseStyleUtils, 'injectBaseStyles')
+      const { rerender } = setup(
+        <Table withBaseStyles={{ breakpoint: '40em' }}>
+          <Thead>
+            <Tr>
+              <Th>Header 1</Th>
+              <Th>Header 2</Th>
+              <Th>Header 3</Th>
+            </Tr>
+          </Thead>
+          <Tbody>
+            <Tr>
+              <Td>Row 1</Td>
+              <Td>Row 2</Td>
+              <Td>Row 3</Td>
+            </Tr>
+          </Tbody>
+        </Table>
+      )
+
+      rerender(
+        <Table withBaseStyles={{ breakpoint: '45em' }}>
+          <Thead>
+            <Tr>
+              <Th>Header 1</Th>
+              <Th>Header 2</Th>
+              <Th>Header 3</Th>
+            </Tr>
+          </Thead>
+          <Tbody>
+            <Tr>
+              <Td>Row 1</Td>
+              <Td>Row 2</Td>
+              <Td>Row 3</Td>
+            </Tr>
+          </Tbody>
+        </Table>
+      )
+
+      expect(baseStyleUtils.injectBaseStyles).toHaveBeenCalledTimes(2)
+    })
+  })
+
+  it('appends custom classes to table elements', () => {
```

Comment:
Enable this functionality such that users can also style the table elements using their own classes or with something like `styled-components`.